gcp/table_gcp_compute_instance_group.go

```diff
@@ -0,0 +1,313 @@
+	if err != nil {
+		return nil, err
+	}
+	project := projectId.(string)
+
+	// Create Service Connection
+	service, err := ComputeService(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Debug("gcp_compute_instance_group.listComputeInstanceGroup", "service_creation_err", err)
+		return nil, err
+	}
+
+	resp := service.InstanceGroups.AggregatedList(project).MaxResults(*pageSize)
+	if err := resp.Pages(ctx, func(page *compute.InstanceGroupAggregatedList) error {
+		for _, item := range page.Items {
+			for _, group := range item.InstanceGroups {
+				d.StreamListItem(ctx, group)
+
+				// Check if context has been cancelled or if the limit has been hit (if specified)
+				// if there is a limit, it will return the number of rows required to reach this limit
+				if d.QueryStatus.RowsRemaining(ctx) == 0 {
+					page.NextPageToken = ""
+					return nil
+				}
+			}
+		}
+		return nil
+	}); err != nil {
+		plugin.Logger(ctx).Debug("gcp_compute_instance_group.listComputeInstanceGroup", "api_err", err)
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getComputeInstanceGroup(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getComputeInstanceGroup")
+
+	// Get project details
+	getProjectCached := plugin.HydrateFunc(getProject).WithCache()
+	projectId, err := getProjectCached(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+	project := projectId.(string)
+
+	var group compute.InstanceGroup
+	name := d.KeyColumnQuals["name"].GetStringValue()
+
+	// Create Service Connection
+	service, err := ComputeService(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Debug("gcp_compute_instance_group.getComputeInstanceGroup", "service_creation_err", err)
+		return nil, err
+	}
+
+	resp := service.InstanceGroups.AggregatedList(project).Filter("name=" + name)
+	if err := resp.Pages(ctx, func(page *compute.InstanceGroupAggregatedList) error {
+		for _, item := range page.Items {
+			for _, i := range item.InstanceGroups {
+				group = *i
+			}
+		}
+		return nil
+	},
+	); err != nil {
+		plugin.Logger(ctx).Debug("gcp_compute_instance_group.getComputeInstanceGroup", "api_err", err)
+		return nil, err
+	}
+
+	// If the specified resource is not present, API does not return any not found errors
+	if len(group.Name) < 1 {
+		return nil, nil
+	}
+
+	return &group, nil
+}
+
+func getComputeInstanceGroupInstances(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	instanceGroup := h.Item.(*compute.InstanceGroup)
+
+	// Get project details
+	getProjectCached := plugin.HydrateFunc(getProject).WithCache()
+	projectId, err := getProjectCached(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+	project := projectId.(string)
+
+	// Create Service Connection
+	service, err := ComputeService(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Debug("gcp_compute_instance_group.getComputeInstanceGroupInstances", "service_creation_err", err)
```

Comment:
```suggestion
		plugin.Logger(ctx).Error("gcp_compute_instance_group.getComputeInstanceGroupInstances", "service_creation_err", err)
```
Should this be error or debug?